src/components/Input/QtySelector/QtySelector.js

```diff
@@ -30,52 +31,80 @@ class QtySelector extends Component {
   static propTypes = {
     disabled: PropTypes.bool,
     value: PropTypes.number,
-    style: PropTypes.objectOf(PropTypes.string)
+    style: PropTypes.objectOf(PropTypes.string),
+    min: PropTypes.number,
+    max: PropTypes.number,
+    onValueUpdated: PropTypes.func
   };
 
   static defaultProps = {
     disabled: false,
     style: {},
-    value: null
+    value: 0,
+    min: QtySelector.MIN_INPUT_VALUE,
+    max: QtySelector.MAX_INPUT_VALUE,
+    onValueUpdated: () => {}
   };
 
   state = {
-    value: this.props.value || "",
+    value: this.props.value,
     focused: false
   };
 
   handleChange = e => {
     const { value } = e.target;
+    const { max, min } = this.props;
 
     if (
       QtySelector.isNumber(value) &&
-      value.length <= QtySelector.MAX_LENGTH_VAL
+      value.length <= QtySelector.MAX_LENGTH_VAL &&
+      value <= max &&
+      value >= min
     ) {
-      this.setState(() => ({ value: Number.parseInt(value, 10) }));
+      this.setState(
+        () => ({ value: Number.parseInt(value, 10) }),
+        () => this.handleValueUpdated()
+      );
     } else if (value === "") {
       // delete input value case
-      this.setState(() => ({ value: "" }));
+      this.setState(() => ({ value: "" }), () => this.handleValueUpdated());
     }
   };
 
-  increment = () =>
-    this.setState(state => {
-      if (state.value === "") {
-        return { value: 0 };
-      }
-      if (state.value === QtySelector.MAX_INPUT_VALUE) {
-        return { value: QtySelector.MAX_INPUT_VALUE };
-      }
-      return { value: state.value + 1 };
-    });
-
-  decrement = () =>
-    this.setState(state => {
-      if (state.value === "" || state.value === 0) {
-        return { value: 0 };
-      }
-      return { value: state.value - 1 };
-    });
+  increment = () => {
+    const { max } = this.props;
+    this.setState(
+      state => {
+        if (state.value === "") {
+          return { value: 0 };
+        }
+        if (state.value === max) {
```

Comment:
We can return early without triggering a state update for this case.

Reply:
Thank you, I have updated this. More information is in this comment: https://github.com/ticketmaster/aurora/pull/581#discussion_r310303065